packages/config/src/proxy.js

```diff
@@ -1,10 +1,19 @@
 const { readFileSync } = require('fs');
 const { sync } = require('glob');
 
-function createInsightsProxy({ betaEnv, rootFolder, localChrome, customProxy = [], publicPath, proxyVerbose, https = true, port }) {
+function createInsightsProxy({ betaEnv, rootFolder, localChrome, customProxy = [], publicPath, proxyVerbose, https = true, port, routes, routesPath }) {
     const target = betaEnv === 'prod' ? 'https://cloud.redhat.com/' : `https://${betaEnv}.cloud.redhat.com/`;
 
-    const isNotCustomContext = (path) => customProxy.length > 0 ? customProxy.reduce((acc, curr) => acc && !curr.context(path), true) : true;
+    let proxyRoutes = routes || (routesPath && require(routesPath));
```

Comment:
What about to change this and the line below to old good `if`, seems too complex to me.

Reply:
Seems a bit too wordy, but I am fine with that
```JS
    let proxyRoutes;
    if (routes) {
        proxyRoutes = routes;
    } else if (routesPath) {
        proxyRoutes = require(routesPath);
    }

    if (proxyRoutes) {
        proxyRoutes = proxyRoutes.routes || proxyRoutes;
    }
```

Reply:
It doesn't have to be.

```
let proxyRoutes = routes;

if(routesPath) {
   const fileRoutes = require(routesPath);
   proxyRoutes = fileRoutes.routes || fileRoutes;
}
```

Reply:
I'd still keep the second if though.
```JS
let proxyRoutes = routes;

if(routesPath) {
   proxyRoutes = require(routesPath);
}

if (proxyRoutes) {
  proxyRoutes = proxyRoutes.routes || proxyRoutes;
}
```

Reply:
I think it would be better to fail instead of pretending that nothing wrong happened if users want to use a file that does not exist.

This is just a specific developer tool, it doesn't have to be fail-proof.